### PR TITLE
Fixes #23310: Port datasources UI to elm

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/23310

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

File: datasources/pom-template.xml [hunk 1]
```diff
@@ -19,12 +19,21 @@
     data sources contacted thought their REST API.
   </description>
 
+  <properties>
+    <destDirectory>datasource</destDirectory>
```
Comment (fanf) [inline datasources/pom-template.xml:23]: the plugin is with a `s`, we should keep it consistant everywhere.

File: datasources/src/main/elm/sources/DataSourceForm.elm [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+                          button [ type_ "button", class "btn btn-danger" ] [--, ng-click "deleteDatasource()", ng-if "!selectedDatasource.isNew",
+                            text "Delete "
+                          , i [ class "fa fa-times-circle" ] []
+                          ]
+                        ]
+                      , div [ class "btn-group" ] [
+                          if (isNew) then text ""
+                          else
+                          button [ type_ "button", class "btn btn-default", onClick (SaveCall { datasource | enabled = not enabled } ) ] [
+                            text (if datasource.enabled then "Enable" else "Disable")
```
Comment (fanf) [inline datasources/src/main/elm/sources/DataSourceForm.elm:61]: it should be the contrary 
image

File: datasources/src/main/elm/sources/APICalls.elm [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    req
+
+saveDataSource : Model -> Bool -> DataSource -> Cmd Msg
+saveDataSource  model isNew datasource =
+  let
+    req =
+      request
+        { method  = if isNew then "PUT" else "POST"
+        , headers = []
+        , url     = getUrl model "datasources"
```
Comment (fanf) [inline datasources/src/main/elm/sources/APICalls.elm:38]: when saving an existing datasource, it should be like delete: `datasources/${id}` 
image

File: datasources/pom-template.xml (hunk 1 above)
Comment (VinceMacBuche) [reply datasources/pom-template.xml:23]: Yep

File: datasources/src/main/elm/sources/APICalls.elm [hunk 4]
```diff
@@ -0,0 +1,62 @@
+module APICalls exposing (..)
+
+
+import Model exposing (DataSource, Model)
```
[the remark this answers is not in the document]
Comment (VinceMacBuche) [reply datasources/src/main/elm/sources/APICalls.elm:4]: The object type is DataSource, so this one should stay like that (but ok, app file should be Datasources.elm and not Datasource.elm

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Pull request merged**